docs/csharp/language-reference/keywords/yield.md

```diff
@@ -72,16 +72,28 @@ On an iteration of the `foreach` loop, the <xref:System.Collections.IEnumerator.
 
 On each subsequent iteration of the `foreach` loop, the execution of the iterator body continues from where it left off, again stopping when it reaches a `yield return` statement. The `foreach` loop completes when the end of the iterator method or a `yield break` statement is reached.
 
-## Example 1
+The following code returns an `IAsyncEnumerable<string>` from an iterator method and then iterates through its elements.
+
+```csharp
+IAsyncEnumerable<string> elements = MyAsyncIteratorMethod();
+await foreach (string element in elements)
+{
+   ...
```

Comment:
```suggestion
   // ...
```

I strive to ensure that code snippets would compile if copied and pasted, I know that `MyAsyncIteratorMethod` doesn't exist, but `...` is even less accurate. I try to leave these as comments.